component/prometheus/remotewrite/types_test.go

```diff
@@ -0,0 +1,55 @@
+package remotewrite
+
+import (
+	"testing"
+
+	"github.com/grafana/agent/pkg/river"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRiverConfig(t *testing.T) {
+	var exampleRiverConfig = `
+		external_labels = {
+			cluster = "local",
+		}
+
+		endpoint {
+			name           = "test-url"
+			url            = "http://0.0.0.0:11111/api/v1/write"
+			remote_timeout = "100ms"
+
+			queue_config {
+				batch_send_deadline = "100ms"
+			}
+		}
+`
+
+	var args Arguments
+	err := river.Unmarshal([]byte(exampleRiverConfig), &args)
+	require.NoError(t, err)
```

Comment:
I guess there is no need to also check the values inside `Arguments`?

Reply:
you are 100% correct that would be even more useful and I'd like to see it. For the timeline and scope of this I am sanity checking the squashing of http_client_config and the call to its validate function on these tests.